app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,20 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new(user_id: params[:user_id], post_id: params[:post_id])
+    if @like.save
+      redirect_to redirect_url, notice: 'You have successfully liked the post'
+    else
+      redirect_to redirect_url, alert: 'Error occurred while liking the post'
+    end
+  end
+
```

Comment:
- [ ] (Optional) Kindly ensure that when you refactor your codebase, you are able to add likes to the currently logged-in user. Currently, this is the case, this is just a reminder to ensure this doesn't break with any refactor.

Example code using the current user to apply likes.
    ...

    def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
	  ...

  end

Reply:
As stated above, as it belongs to current_user method and not included in a project Requirement.

![image](https://github.com/zunairkhan811/blog-app/assets/116083031/6677b939-897e-4ba0-a110-cdedbf64446d)